src/components/post-details-page/PostDetailsPage.js

```diff
@@ -1,46 +1,89 @@
+        //  проверьте лежит ли ваш accessToken в constants/index.js
+        //  прежде чем отправить запрос - включите в true флажок загрузки в стейте isLoading
+        //  выполните fetch запрос за поcтом на `https://gorest.co.in/public-api/posts/${id}?access-token=${accessToken}`
+        //  похожий запрос выполнялся в компоненте PostCard в функции loadComments
+        //  результат выполнения запроса нужно положить в стейт в post
+        //  когда запрос выполнится - не забудьте поменять индикатор загрузки isLoading на false
+        //  обратите внимание, что результат выполнения запроса - ОБЪЕКТ, а не массив
+        const {match: {params: {id}}} = this.props;
+        this.setState({
+            isLoading: true,
+        });
+
+        let response = await fetch(`https://gorest.co.in/public-api/posts/${id}?access-token=${accessToken}`);
+        if (response.ok) {
+            let json = await response.json();
+            const {result} = json;
+            if (result instanceof Object) {
+                this.setState({
+                    isLoading: false,
+                    post: result || null,
+                    error: '',
+                });
+            }
+        } else {
+            this.setState({
+                isLoading: false,
+                error: response.status,
+            });
         }
-      </div>
-    );
-  }
+    };
+
+    render() {
+        //done_todo 3: достать пост из стейта
+        const {post, isLoading, error} = this.state;
+        return (
+            <div>
+                {!!error && <div className="text-center">{error}</div>}
+                {// done_todo 3: если идет загрузка - показываем лоадинг индикатор
+                    //    если нет и пост существует - показываем карточку поста PostCard, в которую как пропсу post передаем наш post из state
+                }
+                {
+                    isLoading && (
+                        <div className="text-center">
+                            <div className="spinner-border text-secondary" role="status">
+                                <span className="sr-only">Loading...</span>
+                            </div>
+                        </div>
+                    )
+                }
+
+                {
+                    !isLoading && post && (
+                        <div className="d-flex justify-content-center">
+                            <PostCard post={post}/>
+                        </div>
+                    )
+                }
+            </div>
+        );
+    }
 }
 
-// // todo 3: обратите внимание - если в App.js вы не передали routerProps в компоненту - то здесь нужно использовать withRouter
-export default PostDetailsPage;
\ No newline at end of file
+// // done_todo 3: обратите внимание - если в App.js вы не передали routerProps в компоненту - то здесь нужно использовать withRouter
+export const PostDetailsPage = withRouter(PostDetailsPageComponent);
```

Comment:
inside App.js you are passing routerProps to the component directly so no need to wrap it in `withRouter `

Reply:
I corrected it.